host/crypto/openssl/cert.c

```diff
@@ -294,49 +274,108 @@ static STACK_OF(X509) * _clone_chain(STACK_OF(X509) * chain)
     return sk;
 }
 
-static oe_result_t _verify_cert(X509* cert_, STACK_OF(X509) * chain_)
+static oe_result_t _verify_cert(
+    X509* cert,
+    STACK_OF(X509) * chain_,
+    const oe_crl_t* const* crls,
+    size_t num_crls)
 {
     oe_result_t result = OE_UNEXPECTED;
     X509_STORE_CTX* ctx = NULL;
-    X509* cert = NULL;
+    X509_STORE* store = NULL;
+    X509* x509 = NULL;
     STACK_OF(X509)* chain = NULL;
 
     /* Clone the certificate to clear any cached verification state */
-    if (!(cert = _clone_x509(cert_)))
-        OE_RAISE(OE_FAILURE);
+    if (!(x509 = _clone_x509(cert)))
+        OE_RAISE_MSG(OE_FAILURE, "Failed to clone X509 cert", NULL);
 
     /* Clone the chain to clear any cached verification state */
     if (!(chain = _clone_chain(chain_)))
-        OE_RAISE(OE_FAILURE);
+        OE_RAISE_MSG(OE_FAILURE, "Failed to clone X509 cert chain", NULL);
+
+    /* Create a store for the verification */
+    if (!(store = X509_STORE_new()))
+        OE_RAISE_MSG(OE_FAILURE, "Failed to allocate X509 store", NULL);
 
     /* Create a context for verification */
     if (!(ctx = X509_STORE_CTX_new()))
-        OE_RAISE(OE_FAILURE);
+        OE_RAISE_MSG(OE_FAILURE, "Failed to create new X509 context", NULL);
 
     /* Initialize the context that will be used to verify the certificate */
-    if (!X509_STORE_CTX_init(ctx, NULL, NULL, NULL))
-        OE_RAISE(OE_FAILURE);
+    if (!X509_STORE_CTX_init(ctx, store, NULL, NULL))
+    {
+        OE_RAISE_MSG(OE_FAILURE, "Failed to initialize X509 context", NULL);
+    }
+
+    /* Create a store with CRLs if needed */
+    if (crls && num_crls)
+    {
+        X509_VERIFY_PARAM* verify_param = NULL;
+
+        for (size_t i = 0; i < num_crls; i++)
+        {
+            crl_t* crl_impl = (crl_t*)crls[i];
+
+            /* X509_STORE_add_crl manages its own addition refcount */
+            if (!X509_STORE_add_crl(store, crl_impl->crl))
```

Comment:
It was a bug. I was tracing it in the debugger and it was leaving the function with more refs on the CRLs than it was entering with since the X509_STORE_add_crl was already managing its own refcount increment. I suspect the original author was not aware of this.

Reply:
Great!